### PR TITLE
Revamp to do expiration check entirely in Dynamo to avoid extra GETs/…

### PR DESCRIPTION
**Pull request by soldiermoth**
…DELETEs; add ability to add a cutoff time; redo tests to match

Previously the Lock's workflow was:
1. GET
2. If expired DELETE
3. PUT conditionally
4. If error sleep RetryWait duration
5. Repeat 1-4 until no error is returned
6. Unlock by DELETE

Updated to 
1. PUT conditionally
2. If error sleep RetryWait duration
3. Repeat 1-2 until no error is returned or reached cutoff time
4. Unlock by DELETE

### CODE REVIEW

**Comment by stuarthicks** [comment]
OK I've stared at this long enough that I'm confident I understand this, and I'm happy with the new behaviour (and it works great in tests elsewhere). However, since I've pushed a bunch of commits to this PR, I'm going to tag it for review from another set of eyes before merging.

File: cmd/ddbsync/main.go
Comment (mjh1) [inline cmd/ddbsync/main.go:11]: is this used when we want to manually unlock or something?

**Review by mjh1 — APPROVED** [review]

File: cmd/ddbsync/main.go
Comment (stuarthicks) [reply cmd/ddbsync/main.go:11]: Could be, but I believe the original purpose is to facilitiate manual testing in addition to the unit tests. At least, that's what I've used it for...

**Pull request merged**
